dbt/adapters/trino/column.py

```diff
@@ -22,4 +24,41 @@ def from_description(cls, name: str, raw_data_type: str) -> "Column":
         # some of the Trino data types specify a type and not a precision
         if raw_data_type.startswith(("array", "map", "row")):
             return cls(name, raw_data_type)
-        return super(TrinoColumn, cls).from_description(name, raw_data_type)
+        match = re.match(r"([^(]+)(\([^)]+\))?([\w ]+)?", raw_data_type)
+        if match is None:
+            raise RuntimeException(f'Could not interpret data type "{raw_data_type}"')
+        data_type, size_info, data_type_2nd_part = match.groups()
+        if data_type_2nd_part:
+            data_type += data_type_2nd_part
+        char_size = None
+        numeric_precision = None
+        numeric_scale = None
+        if size_info is not None:
+            # strip out the parentheses
+            size_info = size_info[1:-1]
+            parts = size_info.split(",")
+            if len(parts) == 1:
+                try:
+                    char_size = int(parts[0])
+                except ValueError:
+                    raise RuntimeException(
+                        f'Could not interpret data_type "{raw_data_type}": '
+                        f'could not convert "{parts[0]}" to an integer'
+                    )
+            elif len(parts) == 2:
+                try:
+                    numeric_precision = int(parts[0])
+                except ValueError:
+                    raise RuntimeException(
+                        f'Could not interpret data_type "{raw_data_type}": '
+                        f'could not convert "{parts[0]}" to an integer'
+                    )
+                try:
+                    numeric_scale = int(parts[1])
+                except ValueError:
+                    raise RuntimeException(
+                        f'Could not interpret data_type "{raw_data_type}": '
+                        f'could not convert "{parts[1]}" to an integer'
+                    )
+
+        return cls(name, data_type, char_size, numeric_precision, numeric_scale)
```

Comment:
This code doesn't add the default values if no precision is given. Should we fix that?

@findinpath, @hovaesco: any opinion on this?

Reply:
@mdesmet I think it does, as this method is called with arguments which are retrieved from running [describe](https://github.com/starburstdata/dbt-trino/blob/master/dbt/include/trino/macros/adapters.sql#L8), so we are passing actual data type (with precision) of column, not the one specified in [dbt config](https://github.com/starburstdata/dbt-trino/blob/09b3bafbb503dec42dae79629dfa6485e1a7c869/tests/functional/adapter/test_seeds_column_types_overrides.py#L85).

Reply:
You are right!